tests/post/steps/test_backup.py

```diff
@@ -0,0 +1,62 @@
+import yaml
+
+from pytest_bdd import scenario, when, then, parsers
+
+from tests import utils
+
+# Scenarios
+@scenario("../features/backup.feature", "Backup multiple times")
+def test_backup(host):
+    pass
+
+
+# When
+@when(parsers.parse("we run the backup script {times} times"))
+def run_backup(request, host, times):
+    iso_root = request.config.getoption("--iso-root")
+
+    for _ in range(int(times)):
+        with host.sudo():
+            res = host.run("%s/backup.sh", str(iso_root))
+            assert res.rc == 0, res.stdout
+
+
+# Then
+@then(parsers.parse("we have {count} backups on each node"))
+def check_backup_archive_count(host, version, ssh_config, count):
+    # using yaml becasue json output is not always correctly formatted
+    get_master_nodes_command = [
+        "salt-run",
+        "--out yaml",
+        "salt.cmd",
+        "metalk8s.minions_by_role",
+        "master",
+        "with_pillar=True",
+    ]
+    master_nodes = yaml.load(
+        utils.run_salt_command(host, get_master_nodes_command, ssh_config).stdout,
+        Loader=yaml.Loader,
```

Comment:
This is needed ?

Reply:
I put a comment about the yaml output in the beginning, 
salt json outputs a single json object for each targetted nod. so output looks like:
`
{
  "attr1": 0 
}
{
  "attr1" 0
}
`

which was too much trouble to parse manually,
since yaml is 'stackable' I preferred to use it